build_functions/CMakeLists.txt

```diff
@@ -207,12 +207,31 @@ function(build_test_folder test_folder)
         (("${test_folder}" MATCHES ".*int.*") AND ${run_int_tests}) OR
         (("${test_folder}" MATCHES ".*perf.*") AND ${run_perf_tests})
     )
+    
+        set(BINARY_DIR) #BINARY_DIR is added to every target (because targets need to have different names).
+        set(PARSING_BINARY_DIR OFF) #see https://cmake.org/cmake/help/latest/command/add_subdirectory.html for "binary_dir"
+        
+        foreach(f ${ARGN}) 
+            set(skip_to_next FALSE)
+            if(${f} STREQUAL "BINARY_DIR")
+                SET(PARSING_BINARY_DIR ON)
+                set(skip_to_next TRUE)
+            endif()
+
+            if(NOT skip_to_next)
```

Comment:
>skip_to_next [](start = 19, length = 12)

I read this a few times and got confused, couldn't this just be an "else" for the condition above?

```
if (string "BINARY_DIR")
    PARSING_BINARY_DIR = true
else if (PARSING_BINARY_DIR)
    BINARY_DIR=...
endif
```

Reply:
I agree that the proposed code looks better and it is easier to read. Changed.

---
In reply to: [458434515](https://github.com/Azure/azure-c-build-tools/pull/32#discussion_r458434515) [](ancestors = 458434515)